profile-widget/diveeventitem.cpp

```diff
@@ -254,20 +255,34 @@ bool DiveEventItem::shouldBeHidden()
 	return false;
 }
 
+int DiveEventItem::depthAtTime(int time)
+{
+	QModelIndexList result = dataModel->match(dataModel->index(0, DivePlotDataModel::TIME), Qt::DisplayRole, time);
+	if (result.isEmpty()) {
+		Q_ASSERT("can't find a spot in the dataModel");
+		hide();
+		return -666;
```

Comment:
The number of the beast as an error number is cute... but I'd prefer something with a symbolic name that is then checked by the caller... because the caller above uses this as a depth (admittedly just to compare against a constant, but still... worst case return "-1" which you define as "ABOVE_SURFACE")

Reply:
I needed some out of bounds special value to signal error (before, when this was part of the other function which is of type void, this was simply a return;.

-1 would not really work (yes, I just realised that the conditional below should take that into account) as that is a valid depth value of a somewhat noisy depth sensor. So I needed a more negative number. But yes, that should be a constant...